vcr/matchers.py

```diff
@@ -30,12 +32,29 @@ def query(r1, r2):
     return r1.query == r2.query
 
 
-def body(r1, r2):
+def raw_body(r1, r2):
     if hasattr(r1.body, 'read') and hasattr(r2.body, 'read'):
         return r1.body.read() == r2.body.read()
     return r1.body == r2.body
 
 
+def body(r1, r2):
+    if hasattr(r1.body, 'read') and hasattr(r2.body, 'read'):
+        r1_body = r1.body.read()
+        r2_body  = r2.body.read()
+    else:
+        r1_body = r1.body
+        r2_body  = r2.body
+    if r1.headers.get('Content-Type') == r2.headers.get('Content-Type') == 'application/x-www-form-urlencoded':
```

Comment:
I would rather not add requests as a dependency.

We also have to be careful here - the requests implementation of CaseInsensitiveDict lowercases the header keys.  Someone might actually care about the capitalization of headers.  Yeah there's something in the RFC that says headers are case-insensitive, but I think keeping VCR transparent is very important.

Since VCR is used mainly for integration tests, it should pass through all the weirdness of the server and client that it's working with, that way it can help find bugs in both.  It shouldn't be helping out by changing headers.

But yeah, the header comparison should probably be case-insensitive for our "smart body matcher".  Just don't store them that way.